File: src/main/java/com/lkimilhol/matchingproject/member/domain/Member.java
package com.lkimilhol.matchingproject.member.domain;

import java.time.LocalDateTime;
import java.util.List;

import javax.persistence.Column;
import javax.persistence.Embeddable;
import javax.persistence.Embedded;
import javax.persistence.Entity;
import javax.persistence.EnumType;
import javax.persistence.Enumerated;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.OneToMany;

import lombok.Getter;
import lombok.NoArgsConstructor;

import com.lkimilhol.matchingproject.address.domain.Address;
import com.lkimilhol.matchingproject.order.domain.Order;

@Entity
@Getter
@NoArgsConstructor
public class Member {

    @Id
    @Column(name = "member_id")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Embedded
    private Nickname nickname;

    private String sex;

    @Embedded
    private Age age;

    @Enumerated(EnumType.STRING)
    private Country country;

    @OneToMany(mappedBy = "member")
    private List<Order> order;

    @Column(name = "update_time", columnDefinition = "DATETIME")
    private LocalDateTime updateTime;

    @Column(name = "insert_time", columnDefinition = "DATETIME")
    private LocalDateTime insertTime;

    public Member(long id) {
        this.id = id;
    }

    private Member (Nickname nickname, String sex, Age age, Country country) {
        this.nickname = nickname;
        this.sex = sex;
        this.age = age;
        this.country = country;
        this.updateTime = LocalDateTime.now();
        this.insertTime = LocalDateTime.now();
    }

    public static Member of(Nickname nickname, String sex, Age age, Country country) {
        return new Member(nickname, sex, age, country);
    }
}

File: src/main/kotlin/com/lkimilhol/matchingproject/member/domain/Member.kt
package com.lkimilhol.matchingproject.member.domain

import com.lkimilhol.matchingproject.order.domain.Order
import java.time.LocalDateTime
import javax.persistence.*

@Entity
class Member (
    @Embedded val nickname: Nickname,
    val gender: Gender,
    @Embedded val age: Age,
    @Enumerated(EnumType.STRING) val country: Country
) {
    @Id
    @Column(name = "member_id")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null

    @OneToMany(mappedBy = "")
    val order: List<Order>? = null

    @Column(name = "update_time", columnDefinition = "DATETIME")
    var updateTime: LocalDateTime? = null

    @Column(name = "insert_time", columnDefinition = "DATETIME")
    var insertTime: LocalDateTime? = null
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as Member

        if (nickname != other.nickname) return false
        if (gender != other.gender) return false
        if (age != other.age) return false
        if (country != other.country) return false
        if (id != other.id) return false
        if (order != other.order) return false
        if (updateTime != other.updateTime) return false
        if (insertTime != other.insertTime) return false

        return true
    }

    override fun hashCode(): Int {
        var result = nickname.hashCode()
        result = 31 * result + gender.hashCode()
        result = 31 * result + age.hashCode()
        result = 31 * result + country.hashCode()
        result = 31 * result + (id?.hashCode() ?: 0)
        result = 31 * result + (order?.hashCode() ?: 0)
        result = 31 * result + (updateTime?.hashCode() ?: 0)
        result = 31 * result + (insertTime?.hashCode() ?: 0)
        return result
    }


}
